frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -382,18 +383,21 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 					d.stage.BaseName = reference.TagNameOnly(ref).String()
 
 					var isScratch bool
-					st, img, err := namedContext(ctx, d.stage.BaseName, dockerui.ContextOpt{ResolveMode: opt.ImageResolveMode.String(), Platform: platform})
+					nc, err := namedContext(ctx, d.stage.BaseName, dockerui.ContextOpt{ResolveMode: opt.ImageResolveMode.String(), Platform: platform})
 					if err != nil {
 						return err
 					}
-					if st != nil {
-						if img != nil {
-							d.image = *img
+					if nc != nil {
+						if nc.Image != nil {
+							d.image = *nc.Image
 						} else {
 							d.image = emptyImage(platformOpt.targetPlatform)
 						}
-						d.state = st.Platform(*platform)
+						d.state = nc.State.Platform(*platform)
 						d.platform = platform
+						// Do not store the named context when it is being used as a base image.
```

Comment:
But if `d.base` is a `namedContext` already then the filter should also be disabled for it.

One way to do it if you also remove the `namedContexts` map would be to initialize `usedPaths` map on setting `dispatchState.namedContext` and then in here if you detect that it is used as base image then set `usedPaths = nil` what would mean that filtering is disabled.

Reply:
I'm not completely certain if this comment is even correct or relevant. I need to check on that today. When I was attempting to test it, I ran into a [panic](https://github.com/moby/buildkit/issues/4177) in the scenario that I was concerned about.

Another way of doing this might be to add `"/"` as a used path when I detect it's being used as a base image. Setting it to nil works, but I feel like it's too much state to keep in your head and too many null pointers that could result in a panic. If we add `"/"` as a used path, then `normalizeContextPaths` will return nil and the local option won't be set.

Reply:
A thing I found to work is disabling the context filtering when the associated dispatch state has commands associated with it. I figure this works well because even if an external stage is using only path `/a.txt`, the stage itself is using the entire base image so I just have commands disabling the filter for named context in general.

When `COPY --from` is used, the implicit stages that are created don't have any commands in them so it's safe to filter them.